Individual Review History - tracking comments, reviewer, and status changes over time

manfromjupyter opened the pull request:
- Closes #204 and is 3rd to last thing before getting reviews fully functional.
- lets screen readers know when certain reviews are missing/were not proposed on all Reviews page.

## Review Roadmap IMO
1. **This PR:**
    - have page or modal users can visit that shows the comments all steps of the way for an individual review, what the status was for each and when as well as all of the changes to the reviewer if there were any.
    - allow comments to be placed every step of the process.
2. **Next PR:** #190 
    - ability for reviewers to opt out of the top 10 if a user was placed on there by another viewer or the system automatically
    - ability to add users manually
    - ability to remove users manually
3. **Eventually:**
    - allow any notebook owner to nominate their notebook for review.
    - turn on numeric ping in the header like change requests when a user is appearing as a recommended reviewer of a notebook.
    - email notification when you the user is manually added to a review, else as part of the subscription email at the start of every day if newly added or when any status changes for any review on notebooks they own (i.e. review is claimed and comment is left)
    - OPTIONAL - feature within app or link to a checklist per type of review what the reviewer needs to review against
    - OPTIONAL (#730) - feature to not lose verified status if it's been reviewed
    - OPTIONAL (#578) - queued reviews stay queued if notebook is updated

If we want to set up ability to turn on allow-lists for certain types of reviews and allow admins to maintain this allow-list for these types of allow-lists if for example compliance reviews can only be done by certain individuals, then we can do this after 3?

manfromjupyter commented:
BLOCKED on this until I can get a `rail generate model` to work with our Github Ci/CD runners or turn it off (which seems to be what's blocking my command or manual creation of it from working)

manfromjupyter commented:
Just noting am no longer blocked but going to focus on the other things now.
```
MySQL [gallery]> show columns from review_history;
```
| Field       | Type         | Null | Key | Default | Extra          |
| ------------- | ------------- |  ------------- | ------------- | ------------- | ------------- |
| id          | int(11)      | NO   | PRI | NULL    | auto_increment |
| review_id   | int(11)      | NO   | MUL | NULL    |                |
| user_id     | int(11)      | YES  | MUL | NULL    |                |
| action      | varchar(191) | NO   |     | NULL    |                |
| reviewer_id | int(11)      | YES  |     | NULL    |                |
| comment     | text         | YES  |     | NULL    |                |

manfromjupyter commented:
**Left to do on this PR**
1. More links back to forth to the review history
2. Ability to view review histories of all reviews from one page OR each review can link to the history OR both?
3. Claim, Unclaim, and Approve all prompt modals to add a comment if the person wishes.

manfromjupyter commented:
@mcrutch to test this on any new database, just need to do a fresh DB migrate and then set config for all 3 types of reviews to in the `settings.yml` to "true".

mcrutch commented:
I feel like maybe this should say "approved" rather than completed even though the only way to get to completed is approved, I just feel like it would be more clear. Thoughts?
![image](https://github.com/nbgallery/nbgallery/assets/51446717/d5e73006-0929-4eba-b408-38c12c8e35c9)
![image](https://github.com/nbgallery/nbgallery/assets/51446717/982946b1-5272-4323-8c06-892a9565ee11)

manfromjupyter commented:
> I feel like maybe this should say "approved" rather than completed even though the only way to get to completed is approved, I just feel like it would be more clear. Thoughts?

@mcrutch I've changed all instances of `completed` to `approved` in our database using a db migrate so other people will also get when they update to latest. I've also changed all logic in our codebase to look for "approved" not "completed" for those review or review_history statuses. Should be ready now.

mcrutch commented on app/views/modals/_review_action.slim:
```diff
@@ -0,0 +1,18 @@
+    div.modal-content
+      div.modal-header
+        h1.modal-title id="reviewActionHeader" Review
+        p.sr-only id="reviewActionDescription" Dialog for adding a comment on claiming, unclaiming, or completing a reviee
```
"reviee"
